data-propagation.md

```diff
@@ -23,12 +26,20 @@ type BlockMsg struct {
 
 The array of message cids must match the `Messages` field in the block when used to construct a [sharray](sharray.md).
 
-Each Filecoin node sets a 'validation function' for the blocks topic that checks that the block is properly constructed, its ticket is valid, the block signature is valid, the miner is a valid miner, and the block is a child of a known good tipset. (TODO: clarify which of these checks are needed, any slowness here impacts propagation time significantly, this is not a full validity check) If an invalid block is received, the peer it was received from should be marked as potentially bad (TODO: we could blacklist peers who send bad blocks, maybe need support from libp2p for this?)
+Every `BlockMsg` received must be validated before being propagated again. If validation fails, it must not be propagated. The validation must check that is its ticket is valid, the block signature is valid, the miner is a valid miner, and the block is a child of a known good tipset.
+
+
+{{% notice todo %}} 
+clarify which of these checks are needed, any slowness here impacts propagation time significantly, this is not a full validity check.
+{{% /notice %}}
 
-TODO: we should likely be smarter here and track which messages we could send along with each block to improve propagation time
 
 ## Message Propagation
 
-Messages are propagated over the libp2p pubsub channel `/fil/messages`. The message is [serialized](data-structures.md#messages) and the raw bytes are sent as the content of the pubsub message.
+Messages are propagated over the libp2p pubsub channel `/fil/messages`. On this channel, every [serialised `Message`](data-structures.md#messages) ist announced.
+
+Upon receiving the message, its validity must be checked: the signature must be valid, and the account in question must have enough funds to cover the actions specified. If the message is not valid it should be dropped and must not be forwarded.
 
-The pubsub validation function for messages checks that the content of each pubsub message on this topic is, first, under the maximum size limit for a message, and then that it is a properly constructed message. (TODO: discuss checking signatures and account balances, some tricky bits that need consideration). If an invalid message is received from a peer, that peer should be marked as potentially bad.
+{{% notice todo %}}
+discuss checking signatures and account balances, some tricky bits that need consideration. Does the fund check cause improper dropping? E.g. I have a message sending funds then use the newly constructed account to send funds, as long as the previous wasn't executed the second will be considered "invalid" ... though it won't be at the time of execution.
```

Comment:
👍 this needs discussion, the scenario you describe here should not result in the second message being dropped.

We should have a spec for the message pool, and this check here should just be 'propagate the message if it gets successfully added to the message pool'

Reply:
> the scenario you describe here should not result in the second message being dropped.

I agree, but the currently mentioned validation function would do that.

> We should have a spec for the message pool, and this check here should just be 'propagate the message if it gets successfully added to the message pool'

What is the check to be put into the message pool though? Because if that is a light-weight test, yes, but if it tried to do a similar introspection/contextual test, we'd just move the point when it was dropped and it would still mean significant delays before we propagate ourselves.

---

I'd rather vote for super slim structure-only validation check before propagating. however, if the client reports back that a transaction failed, it could stop propagating it from a later point. Not sure, if that would actually have any useful in practice.

Reply:
we should move this discussion to the validation-section.